client.go

```diff
@@ -554,6 +554,19 @@ func (c *Client) AddRetryCondition(condition RetryConditionFunc) *Client {
 	return c
 }
 
+// AddRetryAfterErrorCondition adds the basic condition of retrying after enncoutering
+// an error from the http response
+func (c *Client) AddRetryAfterErrorCondition() *Client {
+	c.AddRetryCondition(func(response *Response, err error) bool {
+		if response.IsError() {
```

Comment:
@gtpaulose can you please add a test case for success and failure flow of new condition?
More info: https://codecov.io/gh/go-resty/resty/pull/384/diff?src=pr&el=tree#diff-Y2xpZW50Lmdv

Reply:
Hey did you mean the test case for 'refiring under an error and non error scenario (i.e `response.IsError() == true` and `response.IsError() == false` )? '

Reply:
Yes

Reply:
Alright, ive added the test case. Ive reused an existing path in the test get server switch case, hope thats okay!